packages/host/app/components/operator-mode/container.gts

```diff
@@ -590,9 +607,60 @@ export default class OperatorModeContainer extends Component<Signature> {
   };
 
   @action updateSubmode(submode: Submode) {
+    switch (submode) {
+      case Submode.Interact:
+        this.onCodePathChange(null);
+        break;
+      case Submode.Code:
+        this.onCodePathChange(new URL(this.lastCardInRightMostStack.id));
+        break;
+    }
+
     this.operatorModeStateService.updateSubmode(submode);
   }
 
+  @action onCodePathChange(codePath: URL | null) {
+    this.operatorModeStateService.updateCodePath(codePath);
+    this.loadCardForCodeMode.perform(codePath);
+  }
+
+  loadCardForCodeMode = restartableTask(async (codePath: URL | null) => {
+    try {
+      if (!codePath) return;
+
+      const realmsToCheck = [
+        this.cardService.defaultURL.href,
+        baseRealm.url,
+        ...otherRealmURLs,
+      ];
+      const isInAnyRealms = await Promise.all(
+        realmsToCheck.map(async (realmURL) => {
+          const realmPath = new RealmPaths(realmURL);
+          return realmPath.inRealm(codePath);
+        }),
+      );
+      if (!isInAnyRealms.includes(true))
```

Comment:
this is an awkward test--an array of bool values is a weird thing to return, you should just return the realm that the URL is in. probably this is a side effect of using a Promise.all--but you are doing a promise.all around synchronous code which doesn't make sense.